src/memory_interface.h

```diff
@@ -56,8 +57,12 @@ class MemoryInterface {
     void SetMMIO(MMIORegion& mmio);
     u16 ProgramRead(u32 address) const;
     void ProgramWrite(u32 address, u16 value);
-    u16 DataRead(u16 address); // not const because it can be a FIFO register
-    void DataWrite(u16 address, u16 value);
+    u16 DataRead(u16 address, bool bypass_mmio = false); // not const because it can be a FIFO register
+    void DataWrite(u16 address, u16 value, bool bypass_mmio = false);
+    u16 DataReadA32(u32 address); // not const because it can be a FIFO register
```

Comment:
Actually this can't be a FIFO register, right?

Reply:
You're right, I blindly copied this from the regular `DataRead`. Will fix. (testing the reroute-DMA-back-to-AHB stuff atm)